### DISCUSSION TITLE
What is overrides option in modpack?

### DISCUSSION THREAD

**Discussion by khuongduy354** [gorilla-devs/ferium#334] [Q&A] [answered]
I was reading code in subcommands/modpack/upgrade.rs  and saw

`to_install= read_overrides(temp_dir.join(manifest.name))`

Which then check for all mod files inside temp_dir and install them
the temp_dir is something like ".config/tmp/ferium/..." 
How can the mod files exist in the temp dir at first place? And why is that related to override ?

**Comment by theRookieCoder** [comment, marked as answer]
> How can the mod files exist in the temp dir at first place?

You can see above the line you mentioned;

https://github.com/gorilla-devs/ferium/blob/0304b528b4654dfc920ed7105f9e1c2ee8e507a1/src/subcommands/modpack/upgrade.rs#L156

Here, the contents of the modpack is unzipped to `temp_dir`.

> And why is that related to override?

These override files are config files that the modpack wants you to use. It might be the case that the user has modified some of these configs so they wouldn't want them to be overridden again when they upgrade, which is why the option exists.
